### PR TITLE
feat(new): Added Azure.AppService.AvailabilityZone

### LINKED ISSUE #2964: [RULE] AppService - Deploy zone-redundant App Service Plan
### Existing rule

_No response_

### Suggested rule

An App Service Plan should be configured for multiple Availability Zones, as of Azure Reliability guidance.

### Pillar

Reliability

### Additional context

[ASP-1 - Deploy zone-redundant App Service plans](https://learn.microsoft.com/en-us/azure/reliability/reliability-app-service?tabs=graph%2Ccli#-asp-1---deploy-zone-redundant-app-service-plans)

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2964

Added Azure.AppService.AvailabilityZone.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [ ] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
Ready for review @BernieWhite.

I was initially planning to create the rule for ASE separately, both for customer readability and for our internal readability/handling with the test suite we use. But have not started on it yet before we decide together.

What do you think? Looking forward to your feedback.

**Comment by BernieWhite** [comment]
> Ready for review @BernieWhite.
> 
> I was initially planning to create the rule for ASE separately, both for customer readability and for our internal readability/handling with the test suite we use. But have not started on it yet before we decide together.
> 
> What do you think? Looking forward to your feedback.

Makes sense to me, that would be the preferred approach to have a separate ASE rule.

**Comment by BenjaminEngeset** [comment]
> > Ready for review @BernieWhite.
> > I was initially planning to create the rule for ASE separately, both for customer readability and for our internal readability/handling with the test suite we use. But have not started on it yet before we decide together.
> > What do you think? Looking forward to your feedback.
> 
> Makes sense to me, that would be the preferred approach to have a separate ASE rule.

Great, the other rule will be included in a separate PR.

**Comment by BenjaminEngeset** [comment]
> > Ready for review @BernieWhite.
> > I was initially planning to create the rule for ASE separately, both for customer readability and for our internal readability/handling with the test suite we use. But have not started on it yet before we decide together.
> > What do you think? Looking forward to your feedback.
> 
> Makes sense to me, that would be the preferred approach to have a separate ASE rule.

Great, the other rule will be included in a separate PR.

**Comment by BenjaminEngeset** [comment]
> > Ready for review @BernieWhite.
> > I was initially planning to create the rule for ASE separately, both for customer readability and for our internal readability/handling with the test suite we use. But have not started on it yet before we decide together.
> > What do you think? Looking forward to your feedback.
> 
> Makes sense to me, that would be the preferred approach to have a separate ASE rule.

Great, the other rule will be included in a separate PR.

File: docs/en/rules/Azure.AppService.AvailabilityZone.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+App service plans use auto-scaling to manage the number of instances based on demand. The following rules apply when scaling a zone-redundant app service plan:
+
+- Minimum Instance Count: The app service plan must have a minimum of 3 instances to support zone redundancy.
+ - If you enable availability zones but specify a capacity less than 3, the platform will enforce a minimum instance count of 3.
+- Even Distribution: If the instance count is larger than 3 and divisible by 3, instances are evenly distributed across the three zones.
+- Partial Distribution: Instance counts beyond 3*N are spread across the remaining one or two zones to ensure balanced distribution.
+
+**Important** Configuring zone redundancy with per-application scaling is possible but may increase costs and administrative overhead.
+When `perSiteScaling` is enabled, each application can have its own scaling rules and run on dedicated instances. To maintain zone redundancy, it is crucial that each application’s scaling rules ensure a minimum of 3 instances.
+Without explicitly configuring this minimum, the application may not meet the zone redundancy requirement.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.AppService.AvailabilityZone.md:28]:
```suggestion
App Service plans support zone redundency, which distributes your application running within the plan across Availablity Zones.
Each Availability Zone is a group of phyiscally separated data centers.
Deploying your application with zone redundency:

- Scales your plan to a minimum of 3 instances in a highly available configuration.
  Additional instances can be added manually or on-demand by using autoscale.
- Improves the resiliency against service disruptions or issues affecting a single zone.

Additionally:

- **Even Distribution**: If the instance count is larger than 3 and divisible by 3, instances are evenly distributed across the three zones.
- **Partial Distribution**: Instance counts beyond 3*N are spread across the remaining one or two zones to ensure balanced distribution.

**Important** Configuring zone redundancy with per-application scaling is possible but may increase costs and administrative overhead.
When `perSiteScaling` is enabled, each application can have its own scaling rules and run on dedicated instances.
To maintain zone redundancy, it is crucial that each application’s scaling rules ensure a minimum of 3 instances.
Without explicitly configuring this minimum, the application may not meet the zone redundancy requirement.
```

**Comment by BenjaminEngeset** [comment]
Great feedback @BernieWhite. The requested changes made sense. I have adjusted accordingly now.

Do we need to use the `AllOf` here or will it work as it is now?

**Comment by BernieWhite** [comment]
> Great feedback @BernieWhite. The requested changes made sense. I have adjusted accordingly now.
> 
> Do we need to use the `AllOf` here or will it work as it is now?

No, all good. PowerShell rules will consume all assert statements similar to `AllOf` by default, unless an assert used `return ...` which would abort the remaining code early.

**Comment by BenjaminEngeset** [comment]
Test was failing due to the change in the rule body, I have updated it now @BernieWhite.

Are we ready for merging this now?

**Review by BernieWhite — APPROVED** [review]
All good to merge @BenjaminEngeset. Thanks for the updates.

**Pull request merged**